librz/core/cmd/cmd_analysis.c

```diff
@@ -8812,3 +8740,329 @@ RZ_IPI RzCmdStatus rz_analysis_class_vtable_lookup_handler(RzCore *core, int arg
+		break;
+	default:
+		rz_warn_if_reached();
+		break;
+	}
+
+	if (tbs != core->blocksize) {
+		rz_core_block_size(core, tbs);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analyze_n_bytes_esil_handler(RzCore *core, int argc, const char **argv) {
+	st32 len = core->blocksize, tbs = len;
+
+	len = (st32)rz_num_get(core->num, argv[1]);
+	if (len <= 0) {
+		RZ_LOG_ERROR("Invalid zero or negative arguments.\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	if (len > tbs) {
+		rz_core_block_size(core, len);
+	}
+	core_analysis_bytes_esil(core, core->block, len, 0);
+	if (tbs != core->blocksize) {
+		rz_core_block_size(core, tbs);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analyze_n_bytes_desc_handler(RzCore *core, int argc, const char **argv) {
+	st32 len = core->blocksize, tbs = len;
+
+	len = (st32)rz_num_get(core->num, argv[1]);
+	if (len <= 0) {
+		RZ_LOG_ERROR("Invalid zero or negative arguments.\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	if (len > tbs) {
+		rz_core_block_size(core, len);
+	}
+	core_analysis_bytes_desc(core, core->block, len, 0);
+	if (tbs != core->blocksize) {
+		rz_core_block_size(core, tbs);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analyze_n_bytes_size_handler(RzCore *core, int argc, const char **argv) {
+	st32 len = core->blocksize, tbs = len;
+
+	len = (st32)rz_num_get(core->num, argv[1]);
+	if (len <= 0) {
+		RZ_LOG_ERROR("Invalid zero or negative arguments.\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	if (len > tbs) {
+		rz_core_block_size(core, len);
+	}
+	core_analysis_bytes_size(core, core->block, len, 0);
+	if (tbs != core->blocksize) {
+		rz_core_block_size(core, tbs);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analyze_n_ins_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	st32 l;
+	ut32 count = 1, obs = core->blocksize;
+
+	if (argc > 1) {
+		l = (st32)rz_num_get(core->num, argv[1]);
+		if (l <= 0) {
+			RZ_LOG_ERROR("Invalid zero or negative arguments.\n");
+			return RZ_CMD_STATUS_ERROR;
+		}
+
+		count = l;
+		l *= 8;
+		if (l > obs) {
+			rz_core_block_size(core, l);
+		}
+	}
+
+	switch (state->mode) {
+	case RZ_OUTPUT_MODE_JSON:
+		core_analysis_bytes_json(core, core->block, core->blocksize, count, state->d.pj);
+		break;
+	case RZ_OUTPUT_MODE_STANDARD:
+		core_analysis_bytes_standard(core, core->block, core->blocksize, count);
+		break;
+	default:
+		rz_warn_if_reached();
+		break;
+	}
+
+	if (obs != core->blocksize) {
+		rz_core_block_size(core, obs);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analyze_n_ins_size_handler(RzCore *core, int argc, const char **argv) {
+	ut32 count = 1, obs = core->blocksize;
+	st32 l;
+
+	if (argc > 1) {
+		l = (st32)rz_num_get(core->num, argv[1]);
+		if (l <= 0) {
+			RZ_LOG_ERROR("Invalid zero or negative arguments.\n");
+			return RZ_CMD_STATUS_ERROR;
+		}
+		count = l;
+		l *= 8;
+		if (l > obs) {
+			rz_core_block_size(core, l);
```

Comment:
same here. command takes "number of instructions" but here we modify the block size.